src/NuGet.Core/NuGet.ProjectModel/LockFile/Utf8JsonStreamLockFileTargetLibraryConverter.cs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.ProjectModel
+{
+    /// <summary>
+    /// A <see cref="Utf8JsonStreamReaderConverter{T}"/> to allow read JSON into <see cref="LockFileTargetLibrary"/>
+    /// </summary>
+    internal class Utf8JsonStreamLockFileTargetLibraryConverter : IUtf8JsonStreamReaderConverter<LockFileTargetLibrary>
+    {
+        private static readonly byte[] TypePropertyName = Encoding.UTF8.GetBytes("type");
+        private static readonly byte[] FrameworkPropertyName = Encoding.UTF8.GetBytes("framework");
+        private static readonly byte[] DependenciesPropertyName = Encoding.UTF8.GetBytes("dependencies");
+        private static readonly byte[] FrameworkAssembliesPropertyName = Encoding.UTF8.GetBytes("frameworkAssemblies");
+        private static readonly byte[] RuntimePropertyName = Encoding.UTF8.GetBytes("runtime");
+        private static readonly byte[] CompilePropertyName = Encoding.UTF8.GetBytes("compile");
+        private static readonly byte[] ResourcePropertyName = Encoding.UTF8.GetBytes("resource");
+        private static readonly byte[] NativePropertyName = Encoding.UTF8.GetBytes("native");
+        private static readonly byte[] BuildPropertyName = Encoding.UTF8.GetBytes("build");
+        private static readonly byte[] BuildMultiTargetingPropertyName = Encoding.UTF8.GetBytes("buildMultiTargeting");
+        private static readonly byte[] ContentFilesPropertyName = Encoding.UTF8.GetBytes("contentFiles");
+        private static readonly byte[] RuntimeTargetsPropertyName = Encoding.UTF8.GetBytes("runtimeTargets");
+        private static readonly byte[] ToolsPropertyName = Encoding.UTF8.GetBytes("tools");
+        private static readonly byte[] EmbedPropertyName = Encoding.UTF8.GetBytes("embed");
+        private static readonly byte[] FrameworkReferencesPropertyName = Encoding.UTF8.GetBytes("frameworkReferences");
+
+        public LockFileTargetLibrary Read(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.PropertyName)
+            {
+                throw new JsonException("Expected PropertyName, found " + reader.TokenType);
+            }
+
+            var lockFileTargetLibrary = new LockFileTargetLibrary();
+
+            //We want to read the property name right away
+            var propertyName = reader.GetString();
+            var (targetLibraryName, version) = propertyName.SplitInTwo('/');
+            lockFileTargetLibrary.Name = targetLibraryName;
+            lockFileTargetLibrary.Version = version is null ? null : JsonUtility.ParseNugetVersion(version);
```

Comment:
Separate to this PR, but JsonUtility.ParseNuGetVersion should be a candidate for moving to parsing from a UTF8 string, or at least a ReadOnlySpan<char> to avoid the wasted allocation to split the version into a string, only to throw it away.